### PR TITLE
Support HTTP Bearer Auth

### PR DESCRIPTION
**Pull request by azz**
This PR adds support for HTTP Bearer (Anonymous Token) Auth following the Docker spec:

https://docs.docker.com/registry/spec/auth/token/

This was required to work with an Artifactory docker registry.

### CODE REVIEW

File: resources/plugin.xml [hunk 1]
```diff
@@ -2,15 +2,12 @@
...
         <name>dockerpoller</name>
-        <version>1.1.0</version>
+        <version>1.2.0-PRE</version>
         <target-go-version>17.7.0</target-go-version>
         <description>Docker package poller plugin for Thoughworks GoCD.</description>
         <vendor>
             <name>Magnus Lyckå</name>
             <url></url>
         </vendor>
-        <target-os>
```
Comment (azz) [inline resources/plugin.xml:12]: I tested on a Windows server and it worked.

**Comment by magnus-lycka** [comment]
Hi Lucas,

Thanks for your contribution, and sorry for my slow response.

I've been planning to set up better regression tests, but I realize that I won't have time for that right now, so I'll test this manually. Right now we don't use bearer auth, and I haven't used that, so I'm a bit confused about the change you made.

What I see is that on a 401, you pull a token URL from the realm part of the Www-Authenticate header, and then you get a token from that URL, which you use in a second call to get the actual docker info.
This all seems to match https://docs.docker.com/registry/spec/auth/token/ but the thing that confuses me is why you get a token back!!! I don't see how you are providing any credentials...Or did you just tell Artifactory that anything goes?

**Comment by azz** [comment]
Basically Artifactory is _usually_ anonymous-read by default. You need to log in for all write operations, however the Docker API within Artifactory requires that _all_ calls have a token, even if anonymous read is enabled. So there's no real level of security added, just consistency to the API.

I've been using this build of the plugin for a couple of weeks now with great success!

**Comment by magnus-lycka** [comment]
Gotcha. I would have done it like that too if I had written artifactory. :-)

**Pull request merged**